src/foam/lib/json/Outputter.java

```diff
@@ -96,17 +97,26 @@ protected void outputNull() {
   }
 
   public void outputString(String s) {
+    boolean hasNewLine = s.indexOf('\n') >= 0;
+    if ( multiLineOutput_ && hasNewLine ) writer_.append("\r\n  ");
```

Comment:
I think this would be much more readable as two completely separate code paths, rather than the same conditional every other line.

```
if ( multiLineOutput_ && s.indexOf('\n') >= 0 ) {
  <do multiline output>
  return
}
<existing single-line output code>
```

It's not worth trying to re-use every single shared line such as "writer_.append(escape(s))" at the cost of making the flow difficult to read.  Sometimes a small amount of duplicated code is better.

It's also better to only check indexOf('\n') when multiLineOutput_ is true, otherwise we pay the cost of searching the string every time even when we don't care if its true or not.